simplex/comm.go

```diff
@@ -0,0 +1,143 @@
+		nodeID:     config.Ctx.NodeID[:],
+		logger:     config.Log,
+		sender:     config.Sender,
+		msgBuilder: config.OutboundMsgBuilder,
+		chainID:    config.Ctx.ChainID,
+	}
+
+	return c, nil
+}
+
+// sortNodes sorts the nodes in place by their byte representations.
+func sortNodes(nodes []simplex.NodeID) {
+	slices.SortFunc(nodes, func(i, j simplex.NodeID) int {
+		return bytes.Compare(i, j)
+	})
+}
+
+func (c *Comm) ListNodes() []simplex.NodeID {
+	return c.nodes
+}
+
+func (c *Comm) SendMessage(msg *simplex.Message, destination simplex.NodeID) {
+	outboundMsg, err := c.simplexMessageToOutboundMessage(msg)
+	if err != nil {
+		c.logger.Error("Failed creating message", zap.Error(err))
+		return
+	}
+
+	dest, err := ids.ToNodeID(destination)
+	if err != nil {
+		c.logger.Error("Failed to convert destination NodeID", zap.Error(err))
+		return
+	}
+
+	c.sender.Send(outboundMsg, common.SendConfig{NodeIDs: set.Of(dest)}, c.subnetID, subnets.NoOpAllower)
+}
+
+func (c *Comm) Broadcast(msg *simplex.Message) {
+	for _, node := range c.nodes {
+		if node.Equals(c.nodeID) {
+			continue
+		}
+
+		c.SendMessage(msg, node)
+	}
+}
+
+func (c *Comm) simplexMessageToOutboundMessage(msg *simplex.Message) (message.OutboundMessage, error) {
+	var simplexMsg *p2p.Simplex
+	switch {
+	case msg.VerifiedBlockMessage != nil:
+		bytes, err := msg.VerifiedBlockMessage.VerifiedBlock.Bytes()
+		if err != nil {
+			return nil, fmt.Errorf("failed to serialize block: %w", err)
+		}
+		simplexMsg = newP2PSimplexBlockProposal(c.chainID, bytes, msg.VerifiedBlockMessage.Vote)
+	case msg.VoteMessage != nil:
+		simplexMsg = newP2PSimplexVote(c.chainID, msg.VoteMessage.Vote.BlockHeader, msg.VoteMessage.Signature)
+	case msg.EmptyVoteMessage != nil:
+		simplexMsg = newP2PSimplexEmptyVote(c.chainID, msg.EmptyVoteMessage.Vote.ProtocolMetadata, msg.EmptyVoteMessage.Signature)
+	case msg.FinalizeVote != nil:
+		simplexMsg = newP2PSimplexFinalizeVote(c.chainID, msg.FinalizeVote.Finalization.BlockHeader, msg.FinalizeVote.Signature)
+	case msg.Notarization != nil:
+		simplexMsg = newP2PSimplexNotarization(c.chainID, msg.Notarization.Vote.BlockHeader, msg.Notarization.QC.Bytes())
+	case msg.EmptyNotarization != nil:
+		simplexMsg = newP2PSimplexEmptyNotarization(c.chainID, msg.EmptyNotarization.Vote.ProtocolMetadata, msg.EmptyNotarization.QC.Bytes())
+	case msg.Finalization != nil:
+		simplexMsg = newP2PSimplexFinalization(c.chainID, msg.Finalization.Finalization.BlockHeader, msg.Finalization.QC.Bytes())
+	case msg.ReplicationRequest != nil:
+		simplexMsg = newP2PSimplexReplicationRequest(c.chainID, msg.ReplicationRequest.Seqs, msg.ReplicationRequest.LatestRound)
+	case msg.VerifiedReplicationResponse != nil:
+		msg, err := newP2PSimplexReplicationResponse(c.chainID, msg.VerifiedReplicationResponse.Data, msg.VerifiedReplicationResponse.LatestRound)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create replication response: %w", err)
+		}
+		simplexMsg = msg
+	}
```

Comment:
I think we should add a `default` case to error on. It seems unexpected for us to be sending an empty message.